library/core/src/array/mod.rs

```diff
@@ -533,11 +533,12 @@ impl<T, const N: usize> [T; N] {
     /// assert_eq!(c, Some(a));
     /// ```
     #[unstable(feature = "array_try_map", issue = "79711")]
-    pub fn try_map<F, R>(self, f: F) -> ChangeOutputType<R, [R::Output; N]>
+    pub fn try_map<R>(
+        self,
+        f: impl FnMut(T) -> R,
```

Comment:
Do we have a libs-api policy for using in APITs given the fact that the synthetic type param they get lowered to can't be specified?

Reply:
The closure can be given type annotations.

Reply:
Right, but if one for whatever reason would like to force the closure to be non-capturing for example (quite an artificial one I know), the user can't do so via `try_map::<_, fn(_) -> _>` but need to do `(|…| …) as fn(_) -> _`. I'm talking about practical application here, I'm just curious about T-libs-api's stance on the use of APIT vs. explicit type params in the stdlib going forward (a topic I'm pretty sure has been discussed in the past).

Reply:
Ehn. I think that thinking about it in terms of "APITs" instead of "can someone actually write the damn code?" is a very compiler-brained way to look at it, and that any policy established shouldn't apply equally to a hypothetical random trait versus the `impl Fn*` traits.

Reply:
It's my experience that unnecessary things being in a type signature makes it harder to work with them because now it has to be given as a parameter at every single site, forever and ever, when you actually just wanted to specify the type of `R`.

Reply:
>  T-libs-api's stance on the use of APITs vs. explicit type params in the stdlib going forward (a topic I'm pretty sure has been discussed in the past).

anyways, if T-libs-api has a policy on this, it would be news to me, and they should write it down in, er, the policy section of https://std-dev-guide.rust-lang.org